### PR TITLE
[Task]: Fix phpstan issues

### PR DESCRIPTION
**Pull request by mcop1**
Part of https://github.com/pimcore/pimcore/pull/13889 
The parameter was changed to an array, but this pr was reverted. Also added the native type for `encodeFunc` in `AssetHelperController.php`

Fix issues with `WebDav/Folder` and `Document/Page`

### CODE REVIEW

File: bundles/AdminBundle/src/Controller/Admin/Asset/AssetHelperController.php [hunk 1]
```diff
@@ -765,7 +765,7 @@ public function doExportAction(Request $request, LocaleServiceInterface $localeS
         return $this->adminJson(['success' => true]);
     }
 
-    public function encodeFunc($value): string
+    public function encodeFunc(string $value): string
     {
         $value = str_replace('"', '""', (string) $value);
```
Comment (blankse) [inline bundles/AdminBundle/src/Controller/Admin/Asset/AssetHelperController.php]:
```suggestion
        $value = str_replace('"', '""', $value);
```

**Comment by blankse** [comment]
@mcop1 For the new PhpStan errors from the phpstan dev: I think we should set the https://phpstan.org/config-reference#treatphpdoctypesascertain config to false.

The php docs are correct. The ifs are for wrong argument types.

**Comment by mcop1** [comment]
> @mcop1 For the new PhpStan errors from the phpstan dev: I think we should set the https://phpstan.org/config-reference#treatphpdoctypesascertain config to false.
> 
> The php docs are correct. The ifs are for wrong argument types.

@blankse what do you think about the latest commit? I removed the first `throw`, because if `$asset` is null it should trigger the exception at the end of the method. If you pass e.g. an `int` variable to the method it should the trigger the same exception.

**Comment by blankse** [comment]
@mcop1 LGTM 👍🏼

**Comment by blankse** [comment]
@mcop1 I think this should be also changed for 10.5. I think there the errors also appears when PhpStan released the new version.

**Pull request merged**